gui/webdisplay/src/RWebWindowsManager.cxx

```diff
@@ -154,6 +155,41 @@ void RWebWindowsManager::AssignWindowThreadId(RWebWindow &win)
    }
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////
+/// If LC_ROOT_LISTENER_SOCKET variable is configured,
+/// message will be send to that unix socket
+
+bool RWebWindowsManager::InformListener(const std::string &msg)
+{
+#ifdef R__WIN32
+   (void) msg;
+   return false;
+
+#else
+
+   const char *fname = gSystem->Getenv("ROOT_LISTENER_SOCKET");
+   if (!fname || !*fname)
+      return false;
+
+   TSocket s(fname);
+   if (!s.IsValid())
+      return false;
+
+   int res = s.SendRaw(msg.c_str(), msg.length());
+
+   s.Close();
+
+   if (res > 0) {
+      // workaround to let listener configure port forwarding
+      gSystem->ProcessEvents();
+      gSystem->Sleep(10);
```

Comment:
Why 10 and not 2 or 42? :-) Any chance that we can have a hand-shake, i.e. a second message that conveys "listener is ready"?

Reply:
Seems to be, timeout value is not important. 
It is just `sleep` system call which is matter - it let handle some other system events and let process socket sending.
I can edit comment here - port forwarding is now configured at very beginning automatically. 

By the way, I tried handshaking, but socket configured only in one direction. I need to make another sockets pair, which is not necessary here.

Reply:
OK thanks, all good then. (10 seconds is a lot of time, though.)

Reply:
It is 10 ms.